### PR TITLE
feat: upgrade to Expo SDK 50

### PR DESCRIPTION
**Pull request by frankcalise**
## Please verify the following:

- [x] `yarn test` **jest** tests pass with new tests, if relevant
- [x] `README.md` has been updated with your changes, if relevant

## Describe your PR
- Upgrades to Expo SDK 50 via https://expo.dev/changelog/2024/01-18-sdk-50#%E2%9E%A1%EF%B8%8F-upgrading-your-app
- Refactored tests to accommodate for RN 0.73.x changes of Java -> Kotlin via https://react-native-community.github.io/upgrade-helper/?from=0.72.10&to=0.73.2
- Updated README.md with latest stack versions
- Updated `jest.config.js` to more closely match `jest-expo` default plus our additions
- Upgrades `crashReporting.ts` for Sentry updates with Expo SDK 50
- Disables Flipper
- Engine check for Node 18
- Removes no longer maintained `eslint-plugin-node`

### CODE REVIEW

**Review by mazenchami — APPROVED** [review]
impressive work! 🎉 🚀

**Review by Jpoliachik — APPROVED** [review]
👏

File: boilerplate/jest.config.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ module.exports = {
   ...tsjPreset,
   preset: "jest-expo",
   transformIgnorePatterns: [
-    "<rootDir>/node_modules/(react-clone-referenced-element|@react-native-community|react-navigation|@react-navigation/.*|@unimodules/.*|native-base|react-native-code-push)",
+    "<rootDir>/node_modules/(?!((jest-)?react-native|@react-native(-community)?)|expo(nent)?|@expo(nent)?/.*|@expo-google-fonts/.*|react-navigation|@react-navigation/.*|@unimodules/.*|unimodules|sentry-expo|native-base|react-native-svg|react-clone-referenced-element|react-native-code-push)",
```
Comment (flexbox) [inline boilerplate/jest.config.js:8]: to make it more readable here is a little hack I use

```
const packagesToTransform = [
  "react-native",
  "react-native-(.*)",
  "@react-native",
  "@react-native-community",
  "@react-navigation",
  "expo",
  "expo-(.*)",
  "react-native-svg",
  "@sentry/.*",
  "moti/.*",
  "@shopify/react-native-skia" // https://shopify.github.io/react-native-skia/docs/getting-started/installation#testing-with-jest
];
```

```
  transformIgnorePatterns: [
    `node_modules/(?!(${packagesToTransform.join("|")})/)`
  ],
```

**Comment by flexbox** [comment]
@mattiaferrari02 nice, can you give us a reproductible repository we can dig on? 

And share the stack trace from expo dashboard 🙏

**Comment by frankcalise** [comment]
> I scaffolded a new project from this PR, testing the eas developmnent build does not work. Why did you guys even approve this without trying a build. I hate you all.

Ciao @mattiaferrari02 ! Alcuni punti:
1. This isn't approved
2. Providing the error message you ran into would help us actually attempt to resolve such an issue
3. Please be kind here

Thanks for your understanding

**Comment by frankcalise** [comment]
@mattiaferrari02 can you provide the output from the command `npx ignite-cli@latest doctor` please?

File: boilerplate/jest.config.js (hunk 1 above)
Comment (markrickert) [reply boilerplate/jest.config.js:8]: I like this a lot! So much more readable!

**Pull request merged**